app/_main.py

```diff
@@ -35,13 +39,55 @@
         raise HTTPException(status_code=404, detail="Item not found")
 
 
-@app.get("/questions/{question_id}")
-async def read_question(question_id: int):
-    # keep a memo of questions served, so there are no repeats for a given session
-    # e.g., "questions_served.append({template: 4, game_id: 45})"
-    return {"question_id": question_id}
+@app.get("/questions/")
+async def read_question(asked: str = None):
+    "asked is a comma separated list of question ids that have been asked"
+    # memo passed between client & server ensures questions aren't repeated    
+    asked_memo = set(asked.split(",") if asked is not None else ())
+    (template_index, right_game_id) = get_question_id_parts()
+    new_qid = f"{template_index}-{right_game_id}"
+    while new_qid in asked_memo:
+        (template_index, right_game_id) = get_question_ids()
+        new_qid = f"{template_index}-{right_game_id}"
+    return {
+        "question": Question(
+            id=new_qid,
+            template_index=template_index,
+            correct_game=all_games_by_id[right_game_id],
+            answers=get_answers(template_index,right_game_id)
+        ), 
+        "asked": asked
+    }
 
 
 @app.post("/score/")
-async def create_item(score: Score):
-    return score
+async def create_item(player_name: str, score: int):
+    return Score(player_name, score)
+
+# helpers
+def get_question_id_parts():
+    template_index = random.randrange(len(question_templates))
+    right_game_id = random.choice(all_game_ids)
+    return template_index, right_game_id
+
+def get_answers(template_index,right_game_id):
+    template = question_templates[template_index]
+    right_game = all_games_by_id[right_game_id]
+    right_answer = getattr(right_game, template["answer_type"])
+    answers = set()
+    answers.add(oxford_join(right_answer))
+    while len(answers) < 4:
+        if template["answer_type"] == "year_published":
+            # is the complication of code worth the extent to which this is 
+            # better than sampling random games?
+            alt_answer = random.randint(
+                right_game.year_published-10, 
+                min(datetime.now().year, right_game.year_published+10)
+            )
+        else:
+            alt_answer = getattr(
+                random.choice(all_games), 
+                template["answer_type"]
+            )
+        answers.add(oxford_join(alt_answer))
+    return list(answers)
```

Comment:
Doesn't this imply that the right answer is always first in the list? Seems like you might want to avoid that, making the server responsible for randomization instead of the client.